### PR TITLE
release-21.1: Add internal tables for detecting lost table data and clearing it

### PR DESCRIPTION
**Pull request by fqazi**
Backport:
  * 1/1 commits from "sql: add an internal table showing lost table descriptors" (#62801)
  * 1/1 commits from "sql: builtin that clears a range of table data based on descID" (#62620)

Please see individual PRs for details.

/cc @cockroachdb/release

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/schema_repair [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+select * from crdb_internal.force_delete_table_data($t_id)
+----
+true
+
+statement ok
+select * from crdb_internal.unsafe_upsert_descriptor($t_id, crdb_internal.json_to_pb( 'cockroach.sql.sqlbase.Descriptor','$json'))
+
+query I
+SELECT * FROM forcedeletemydata ORDER BY v ASC
+----
```
Comment (postamar) [inline pkg/sql/logictest/testdata/logic_test/schema_repair:189]: For some reason all these logic tests end up in the first of your two commits.

**Review by postamar — APPROVED** [review]
LGTM modulo my comment about logic tests. I'm not sure if this matters, though, it's up to you.

File: pkg/sql/logictest/testdata/logic_test/schema_repair (hunk 1 above)
Comment (fqazi) [reply pkg/sql/logictest/testdata/logic_test/schema_repair:189]: Yeah, not sure why that happened this was done via backport. It should have been spread between the two commits and I saw signs of that happening with the merge conflicts. I think it's harmless, and not worth digging into and I'll merge once CI comes back.

Comment (postamar) [reply pkg/sql/logictest/testdata/logic_test/schema_repair:189]: I suppose it only matters if one would want to revert the second of the two commits, but not the first.

**Pull request merged**